pkg/backend/mock.go

```diff
@@ -519,3 +519,38 @@ func (ms *MockStack) DefaultSecretManager(info *workspace.ProjectStack) (secrets
 	}
 	panic("not implemented")
 }
+
+//
+// Mock stack reference
+//
+
+// MockStackReference is a mock implementation of [StackReference].
+// Set the fields on this struct to control the behavior of the mock.
+type MockStackReference struct {
+	StringF             func() string
```

Comment:
Might be simpler to just make this plain fields (i.e. just string) and panic if they're ever `""`?